### ISSUE TITLE
[java] UnusedPrivateMethod  false positive when method reference has no target type

### ISSUE THREAD

**Issue by anuragagarwal561994** [pmd/pmd#5083]
**Affects PMD Version:** 7.x

**Rule:** UnusedPrivateMethod

Please provide the rule name and a link to the rule documentation:
https://docs.pmd-code.org/pmd-doc-7.2.0/pmd_rules_java_bestpractices.html#unusedprivatemethod

**Description:**
When using static private methods as lambda in enum, it treats the methods as unused while it is still being used in enum declaration

**Code Sample demonstrating the issue:**

```java
@Getter
@RequiredArgsConstructor
enum GenerationType {
    APPLE_DESKTOP("https://apps.apple.com/app/id", GenerationType::isAppleType),
    APPLE_ITUNES("https://itunes.apple.com/app/id", GenerationType::isAppleType),
    SAMSUNG("https://www.samsung.com/us/appstore/app/", GenerationType::isSamsungType),
    ROKU("https://channelstore.roku.com/details/", GenerationType::isRokuType),
    AMAZON("https://www.amazon.com/dp/", GenerationType::isAmazonType),
    ANDROID("https://play.google.com/store/apps/details?id=", GenerationType::isAndroidType);

    private final String baseUrl;
    private final Predicate<String> predicate;

    private static boolean isAppleType(String data) {
        return "apple".equals(data);
    }

    private static boolean isRokuType(String data) {
        return "roku".equals(data);
    }

    private static boolean isSamsungType(String data) {
        return "samsung".equals(data);
    }

    private static boolean isAmazonType(String data) {
        return "amazon".equals(data);
    }

    private static boolean isAndroidType(String data) {
        return "android".equals(data);
    }
}
```

**Expected outcome:**
The above methods should not be reported as being usunsed.

**Running PMD through:** Maven

**Comment by jsotuyod** [comment]
This can be reproduced when the code is not compiled first or the class in not included in the auxclasspath, so we don't see the lombok-generated constructor, and the symbol is unresolved.

In this sense, the underlaying issue is the same as with #5097, the auxclasspath is incomplete, and PMD is unable to resolve some symbols.

For this particular rule however, the use of symbol resolution is needed both for overloads, and to detect cases where the only calls to a private method are recursive calls, and therefore, the code is never actually executed. The rule right now errs on false positives, but excluding cases of unresolved symbols would err on false negatives such as this code reporting 0 violations:

```java
public class Foo {
    private void foo(AnotherUnresolvedType baz) {}
    private void foo(UnresolvedType baz) {}
    public void bar(Set<UnresolvedType> s) {
        s.forEach(this::foo);  // Not being able to resolve UnresolvedType and AnotherUnresolvedType means we can't tell which foo is being used here…
    }
}
```

The only way to ensure PMD reports accurate results is running it with a complete auxclasspath. In this case that means having compiled the code.

Alternatively, this could be a good usage of #4327

**Comment by anuragagarwal561994** [comment]
Correct me if I am wrong, but I am using pmd maven plugin to execute in the very phase which is after everything has been compiled and done so why it should not be part of my classpath or understand the generated code?
